urllib3/util/ssl_.py

```diff
@@ -17,6 +18,26 @@
 
 import errno
 import warnings
+import hmac
+
+if hasattr(hmac, 'compare_digest'):
+    _const_compare_digest = hmac.compare_digest
+else:
+    def _const_compare_digest(a, b):
+        """
+        Compare two digests of equal length in constant time.
+
+        Returns True if the digests match, and False otherwise.
+        """
+        if len(a) != len(b):
+            raise ValueError("Mismatched input lengths!")
+
+        result = 0
+        for l, r in six.moves.zip(bytearray(a), bytearray(b)):
```

Comment:
The semantics of `zip` are different in Python 2 and 3, as far as I understand. In Python 2, it will pad the sequence with None values, whereas in Python 3 it will stop iterating when an input argument gets exhausted. That could result in 'abc' and 'ab' deemed matching, which is confusing and not the behavior of the built-in function we are trying to replace. I realize this shouldn't happen because different lengths should be impossible, but I don't think not worrying about six is worth this.